### PR TITLE
Deprecate xpack and typed endpoints for graph explore api

### PR DESCRIPTION
**Pull request by pgomulka**
Previously the `/{index}/{type}/_xpack/graph/_explore` api was indicating to use `/{index}/{type}/_graph/explore` which is also deprecated.
This commit deprecates all typed endpoints

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/rest/RestController.java [hunk 1]
```diff
@@ -253,7 +253,7 @@ private void dispatchRequest(RestRequest request, RestChannel channel, RestHandl
             if (handler.allowsUnsafeBuffers() == false) {
                 request.ensureSafeBuffers();
             }
-
+            //TODO why we are using client.threadPool().getThreadContext() ?
```
Comment (pgomulka) [inline server/src/main/java/org/elasticsearch/rest/RestController.java:256]: I wonder if we should not be using the thread context passed over via `AbstractHttpServerTransport#dispatchRequest`
The thread pool is first created in node and is the used in both `RestController` and `AbstracthttpServerTransport`

File: test/framework/src/main/java/org/elasticsearch/test/rest/RestActionTestCase.java [hunk 2]
```diff
@@ -66,7 +66,10 @@ protected RestController controller() {
     protected void dispatchRequest(RestRequest request) {
         FakeRestChannel channel = new FakeRestChannel(request, false, 1);
         ThreadContext threadContext = new ThreadContext(Settings.EMPTY);
-        controller.dispatchRequest(request, channel, threadContext);
```
Comment (pgomulka) [inline test/framework/src/main/java/org/elasticsearch/test/rest/RestActionTestCase.java:69]: this in fact was not really chaning the context used in `RestController` - not in all places.
In same places the thread context fetched from thread pool was used

File: server/src/main/java/org/elasticsearch/rest/RestController.java (hunk 1 above)
Comment (jaymode) [reply server/src/main/java/org/elasticsearch/rest/RestController.java:256]: OOF! Good catch. This is a mistake IMO; we should do one of the following things:

1. Remove ThreadContext from the request dispatcher API and always use the one from the node client
2. Leave the API alone and pass the ThreadContext from the dispatcher to all the other methods
3. Add an assertion in dispatch*Request methods that assert the passed threadcontext is the same as that on the node client

This mainly is just an issue for testing but it is tricky and would be nice if we just had one way to access the value

File: test/framework/src/main/java/org/elasticsearch/test/rest/RestActionTestCase.java [hunk 3]
```diff
@@ -66,7 +66,10 @@ protected RestController controller() {
     protected void dispatchRequest(RestRequest request) {
         FakeRestChannel channel = new FakeRestChannel(request, false, 1);
         ThreadContext threadContext = new ThreadContext(Settings.EMPTY);
```
Comment (jaymode) [inline test/framework/src/main/java/org/elasticsearch/test/rest/RestActionTestCase.java:68]: no need to create this anymore

```suggestion
```

**Review by joegallo — APPROVED** [review]

**Pull request merged**